msu/systems/persistent_data/persistent_data_system.nut

```diff
@@ -0,0 +1,109 @@
+::MSU.Class.PersistentDataSystem <- class extends ::MSU.Class.System
+{
+	Mods = null;
+	ModConfigPath = "mod_config/";
+
+	constructor()
+	{
+		base.constructor(::MSU.SystemID.PersistentData);
+		this.Mods = {};
+	}
+
+	function registerMod( _mod )
+	{
+		base.registerMod(_mod);
+
+		_mod.PersistentData = ::MSU.Class.PersistentDataModAddon(_mod);
+		this.addMod(_mod.getID());
+		this.importModConfigFiles(_mod.getID());
+	}
+
+	function importModConfigFiles( _modID )
+	{
+		local persistentDirectory = this.IO.enumerateFiles(this.ModConfigPath + _modID);
+		if (persistentDirectory == null)
+		{
+			return;
+		}
+		foreach (file in persistentDirectory)
+		{
+			local components = split(file, "/");
+			local modID = components[1];
+			local settingsType = components[2];
+			::MSU.Mod.Debug.printWarning(format("Checking file, potential modID: '%s' and settingstype '%s'.", modID, settingsType), "persistence");
```

Comment:
Pretty sure this will throw since mods aren't given a persistence debug flag when they register with this system.

Reply:
It prints from the MSU mod though